### DISCUSSION TITLE
Toggle-Event for Mooney M20R Altitude Preselector?

### DISCUSSION THREAD

**Discussion by johans1234** [nguyenquyhy/Flight-Tracker-StreamDeck#141] [Q&A] [answered]
most of the cockpit functions are well supported - great job !!!! 

but...
My favorite plane is the **Carenado M20R Ovation**. This plane has a very unique instrument to preselect and arm the target altitude and to select the vertical speed (if you want to change the standard of +-500 fpm). I searched the API documentation up and down, but I can`t find a solution to use my stream deck to operate this instrument. Is there a way to do this? (even with a keyboard shortcut, but I can't find such either)

**Comment by rmroc451** [comment]
I'd recommend downloading and running FSUIPC and doing the actions in the cockpit that you are looking to implement, with the console log of FSUIPC open.

If it registers an event in the log, you can check the struct file to see if we have it and if not, raise an issue and we can add it.

I don't have the Mooney, or I'd check for you!

**Reply by johans1234** [reply]
perfect. that´s exactly the approach I was thinking of, I just didn't know what tool I have to use. I will try tomorrow and let you know, if I could figure out the event triggered by the instrument button.

**Comment by brlowe1965** [comment]
I have been getting ready to create one of those for an air manager panel I'm creating for the mooney so if I find them I will let you know. If you find them please post also.

**Comment by brlowe1965** [comment, marked as answer]
Ok I have never used FSUIPC7 so I took a quick stab at it and found the following being used when selecting the ARM button and selecting altitude before the sim CTD.
AP_PANEL_ALTITUDE_ON
AP_VS_VAR_SET_ENGLISH
AP_ALT_VAR_SET_ENGLISH
These may at least get you into the right area to find the info in the SDK.

**Reply by rmroc451** [reply]
nice! @nguyenquyhy  just added those to our struct file ~10 days ago: 

https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/commit/2c5980764be142dc890a911d8700b597e8e49dde

**Reply by brlowe1965** [reply]
Please share what you come up with when your done.

**Reply by brlowe1965** [reply]
By the way every time I loaded FSUIPC7 and ran it my Sim crashed in about 5 minutes or less. I turned off all addon's

**Reply by johans1234** [reply]
mhh, if I run FSUIPC7, I just get the following, when I press the ARM button on the preselector instrument:

AP_ALT_HOLD_OFF (if "hold altitude" was set on the autopilot panel before)
AP_PANEL_ALTITUDE_OFF (if "hold altitude" was not set on the autopilot panel before)

This button is a toggle button and I always get the above events in the log file. But if I put these events as "toggle event" in the Streamdeck plugin, nothing happens in the cockpit of the Mooney - the button doesn`t move if I press the Stream Deck button. 

If I dial the rotary button to select the target altitude, I get nothing in the log. I am completely new with FSUIPC7, so maybe I did something wrong or missed something to configure within this "tool".

**Reply by brlowe1965** [reply]
Try toggling the button on the sim panel a few times very quickly on and off so you have a few in the log near each other. I'm new to FSUIPC7 myself so I do not have all the answers. In the log when your spinning the alt select knob it may be that you and I are not monitoring the correct info to see that value change. I wish I could turn off all the log entries for nav and com as that just fills up the log file.

**Reply by brlowe1965** [reply]
johans1234 did you have any luck with the altitude selector?

**Reply by helibrewer** [reply]
That Altitude preselector is the KAS 297B. I have found that if I use the Preset Toggle Button in the plug-in and select Altitude for the function, it behaves as expected, however there is no feedback in the KAS297B, it just remains at 0,000. The same holds for Vertical Speed, use the Vertical Speed Function in the Preset Toggle Button. You can also use the "Increase value" and "Decrease value" buttons to change the ALT and VS values. This is a work-around since it does not appear to be using any of the KAS297B functionality. 
I tried these Event ID's also:
MobiFlight.KAP140_Push_ARM
MobiFlight.KAP140_Knob_Inner_INC
MobiFlight.KAP140_Knob_Inner_DEC
MobiFlight.KAP140_Knob_Outer_INC
MobiFlight.KAP140_Knob_Outer_DEC
but the Mooney uses the KAP150 Autopilot so these KAP140 events do not seem to do anything.

**Reply by johans1234** [reply]
thanks. sometimes it is so easy. this is actually exactly what I was looking for - nevertheless, it would be even better, if the KAS 297B would display the values
